File: tutorials/testing/java/shape_collection.java
import bridges.connect.Bridges;
import bridges.base.SymbolCollection;
import bridges.base.Rectangle;
import bridges.base.Circle;
import bridges.base.Text;
import bridges.base.Polyline;
import bridges.base.Polygon;
import bridges.base.Color;
import bridges.base.SymbolGroup;


public class shape_collection {

	public static void main(String[] args) throws Exception {

		//create Bridges object
		Bridges bridges = new Bridges(YOUR_ASSIGNMENT_NUMBER, "YOUR_USER_ID",
			"YOUR_API_KEY");
		// title, description
		bridges.setTitle("Symbol Collection");
		bridges.setDescription("Red square, green circle, magenta horizontal and vertical lines, "
			+	"and a test label with a purple outline.");

		// create some symbols and add to symbol collection
		SymbolCollection sc = new SymbolCollection();

		// draw a rectangle
		Rectangle s1 = new Rectangle(-25.0f, 0.0f, 50.0f, 50.0f);
		s1.setFillColor("red");
		s1.setOpacity(0.3f);
		sc.addSymbol(s1);

		// and a circle next to it
		Circle s2 = new Circle(25.0f, 0.0f, 25.0f);
		s2.setFillColor("green");
		s1.setOpacity(0.1f);
		sc.addSymbol(s2);

		// draw a triangle above
		Polygon s3 = new Polygon();
		s3.addPoint(-25.0f, -10.0f);
		s3.addPoint(25.0f, -10.0f);
		s3.addPoint(0.0f, 50.0f);
		s3.setStrokeColor("darkorchid");
		s3.setStrokeWidth(2.0f);
		sc.addSymbol(s3);

		// draw axes
		Polyline s4 = new Polyline();
		s4.addPoint(-100.0f, 0.0f);
		s4.addPoint(100.0f, 0.0f);
		s4.setStrokeColor("magenta");
		s4.setStrokeWidth(2.0f);
		sc.addSymbol(s4);

		Polyline s5 = new Polyline();
		s5.addPoint(0.0f, -100.0f);
		s5.addPoint(0.0f, 100.0f);
		s5.setStrokeWidth(2.0f);
		sc.addSymbol(s5);

		Polyline s5b = new Polyline();
		s5b.addPoint(50.0f, -100.0f);
		s5b.addPoint(50.0f, 100.0f);
		s5b.addPoint(70.0f, 120.0f);
		s5b.setStrokeWidth(2.0f);
		sc.addSymbol(s5b);
		
		Polygon s6 = new Polygon();
		s6.addPoint (-30.0f, 40.0f);
		s6.addPoint (30.0f, 40.0f);
		s6.addPoint (0.0f, 80.0f);
		s6.setStrokeColor("cyan");
		s6.setFillColor("yellow");
		s6.setStrokeWidth(3.0f);
		sc.addSymbol(s6);



		Text l = new Text();
		l.setAnchorLocation(0.0f, 25.0f);
		l.setFontSize(18f);
		l.setStrokeWidth(2.0f);
		l.setOpacity(.3f);
		l.setText("test label1");
		l.setStrokeColor("purple");
		sc.addSymbol(l);


		SymbolGroup group = new SymbolGroup();
		sc.addSymbol(group);
		group.setTransform(1,0,0,1,30,20);
		
		
		Text l2 = new Text();
		l2.setAnchorLocation(0.0f, 25.0f);
		l2.setFontSize(12f);
		l2.setStrokeWidth(1.0f);
		l2.setText("test label translated and scaled");
		l2.setStrokeColor("purple");
		l2.scale(2.0f, 1.0f);
		group.addSymbol(l2);
		
		SymbolGroup group2 = new SymbolGroup();
		group.addSymbol(group2);
		group2.setTransform(1,0,0,1,30,20);
		group2.rotate(20);
		
		Text l3 = new Text();
		l3.setAnchorLocation(0.0f, 25.0f);
		l3.setFontSize(12f);
		l3.setStrokeWidth(1.0f);
		l3.setText("test label translated more");
		l3.setStrokeColor("purple");
		l3.setAnchorAlignment("top", "middle");
		group2.addSymbol(l3);
		
		
		// set visualizer type
		bridges.setDataStructure(sc);

		// visualize the JSON and Collection
		bridges.visualize();
	}
}

File: tutorials/testing/c++/shape_collection.cpp
#include "Bridges.h"
#include "SymbolCollection.h"
#include "SymbolGroup.h"
#include "Rectangle.h"
#include "Polyline.h"
#include "Polygon.h"
#include "Circle.h"
#include "Text.h"

using namespace bridges;

int main(int argc, char **argv) {
	// create Bridges object
	Bridges bridges (YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID",
		"YOUR_API_KEY");

	// title, description
	bridges.setTitle("Symbol Collection");
	bridges.setDescription("Red square, green circle, magenta horizontal and vertical lines, and a test label with a purple outline.");


	// create some symbols and add to symbol collection
	SymbolCollection sc;

	std::shared_ptr<Rectangle> r1 = std::make_shared<Rectangle>(-50, 0, 25, 25);

	r1->setFillColor(Color("red"));
	r1->setOpacity(0.1f);
	sc.addSymbolPtr(r1);
	

	std::shared_ptr<Rectangle> s1 = std::make_shared<Rectangle>(-50, 0, 25, 25);

	s1->setFillColor(Color("yellow"));
	s1->setOpacity(0.1f);
	
	// and a circle next to it
	Circle s2(50, 0, 25);
	s2.setFillColor(Color("green"));
	s2.setOpacity(1.0f);

	// create a symbol group and put s1 and s2 into it
	SymbolGroup g;
		g.translate(3., 10.);
		g.addSymbolPtr(s1);
		g.addSymbol(s2);

	// add the group in the symbol collection 
	SymbolCollection sc2;
		sc2.addSymbol (g);

	Polygon s3;
	s3.addPoint(-25, -10);
	s3.addPoint(25, -10);
	s3.addPoint(0, 50);
	s3.setStrokeColor("darkorchid");
	s3.setStrokeWidth(2.0f);
	s3.setOpacity(1.0f);
	sc.addSymbol(s3);

	// draw axes
	Polyline s4;
	s4.addPoint(-100, 0);
	s4.addPoint(100, 0);
	s4.setStrokeColor("magenta");
	s4.setStrokeWidth(2.0f);
	s4.setOpacity(1.0f);
	sc.addSymbol(s4);

	Polyline s5;
	s5.addPoint(0, -100);
	s5.addPoint(0, 100);
	s5.setStrokeWidth(2.0f);
	s5.setOpacity(1.0f);
	sc.addSymbol(s5);

	Polygon s6;
	s6.addPoint (-30.0f, 40.0f);
	s6.addPoint (30.0f, 40.0f);
	s6.addPoint (0.0f, 80.0f);
	s6.setStrokeColor("cyan");
	s6.setFillColor("yellow");
	s6.setStrokeWidth(3.0f);
	s6.setOpacity(1.0f);
	sc.addSymbol(s6);

	// an example text label
	Text l;
	float loc[2] = {0, 25};
	l.setAnchorLocation(loc);
	l.setFontSize(50.0f);
	l.setStrokeWidth(1.0f);
	l.setText("test label");
	l.setStrokeColor("purple");
	l.setOpacity(1.0f);
	sc.addSymbol(l);

	sc.addSymbol(g);

	// set visualizer type
	bridges.setDataStructure(sc);

	// visualize the JSON and Collection
	bridges.visualize();

	return 0;
}
